src/typing_extensions.py

```diff
@@ -111,17 +111,23 @@ def _check_generic(cls, parameters, elen=_marker):
                         f" actual {alen}, expected {elen}")
 
 
+def _is_generic_alias_like(t) -> bool:
+    return hasattr(t, "__args__") and hasattr(t, "__origin__") and hasattr(t, "__parameters__")
+
+
 if sys.version_info >= (3, 10):
     def _should_collect_from_parameters(t):
         return isinstance(
             t, (typing._GenericAlias, _types.GenericAlias, _types.UnionType)
-        )
+        ) or _is_generic_alias_like(t)
```

Comment:
This is the only place I put this because this fixes the immediate issue that I know of. There are of course other references to `GenericAlias` but I refrained from applying this everywhere without a good test / use case.